profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/features/VolumeApi.java

```diff
@@ -19,20 +19,118 @@
 import org.apache.jclouds.profitbricks.rest.domain.Volume;
+import org.apache.jclouds.profitbricks.rest.domain.options.DepthOptions;
+import org.apache.jclouds.profitbricks.rest.util.ParseId;
+import org.jclouds.Fallbacks;
+import org.jclouds.Fallbacks.EmptyListOnNotFoundOr404;
 import org.jclouds.http.filters.BasicAuthentication;
 import org.jclouds.http.functions.ParseJson;
 import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PATCH;
+import org.jclouds.rest.annotations.PayloadParam;
 import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.util.Strings2;
 
-@Path("/volumes")
+@Path("/datacenters/{dataCenterId}/volumes")
 @RequestFilters(BasicAuthentication.class)
 public interface VolumeApi extends Closeable {
    
+   @Named("volume:list")
+   @GET
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<Volume> getList(@PathParam("dataCenterId") String dataCenterId);
+
+   @Named("volume:list")
+   @GET
+   @SelectJson("items")
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<Volume> getList(@PathParam("dataCenterId") String dataCenterId, DepthOptions options);
+   
+   @Named("volume:get")
+   @GET   
+   @Path("/{volumeId}")
+   @ResponseParser(VolumeApi.VolumeParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Volume getVolume(@PathParam("dataCenterId") String dataCenterId, @PathParam("volumeId") String volumeId);
+
+   @Named("volume:get")
+   @GET   
+   @Path("/{volumeId}")
+   @ResponseParser(VolumeApi.VolumeParser.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Volume getVolume(@PathParam("dataCenterId") String dataCenterId, @PathParam("volumeId") String volumeId, DepthOptions options);
+   
+   @Named("volume:create")
+   @POST
+   @MapBinder(CreateVolumeRequestBinder.class)
+   @ResponseParser(VolumeApi.VolumeParser.class)
+   Volume createVolume(@PayloadParam("volume") Volume.Request.CreatePayload payload);
+   
+   @Named("volume:update")
+   @PATCH
+   @Path("/{volumeId}")
+   @MapBinder(UpdateVolumeRequestBinder.class)
+   @ResponseParser(VolumeApi.VolumeParser.class)
+   @Produces("application/vnd.profitbricks.partial-properties+json")
+   Volume updateVolume(@PayloadParam("volume") Volume.Request.UpdatePayload payload);
+   
+   @Named("volume:delete")
+   @DELETE
+   @Path("/{volumeId}")
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   void deleteVolume(@PathParam("dataCenterId") String dataCenterId, @PathParam("volumeId") String volumeId);
+   
+   @Named("volume:snapshot:create")
+   @POST
+   @MapBinder(CreateSnapshotRequestBinder.class)
+   @ResponseParser(SnapshotApi.SnapshotParser.class)
+   Snapshot createSnapshot(@PayloadParam("snapshot") Volume.Request.CreateSnapshotPayload payload);
```

Comment:
Hmm. Should this (and `restoreSnapshot`) be in `SnapshotApi.java` instead? If so, don't forget to move both of their [tests](https://github.com/jclouds/jclouds-labs/pull/233/files#diff-81aadea0059ec0e66cf6304f6234762bR148) too!